units/commit_queue.go

```diff
@@ -306,23 +306,29 @@ func (j *commitQueueJob) processGitHubPRItem(ctx context.Context, cq *commitqueu
 		j.logError(err, "can't insert patch", nextItem)
 		j.dequeue(cq, nextItem)
 		j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't make patch", ""))
+		return
 	}
 	nextItem.Version = patchDoc.Id.Hex()
 	if err = cq.UpdateVersion(nextItem); err != nil {
 		j.logError(err, "problem saving version", nextItem)
+		j.dequeue(cq, nextItem)
+		j.AddError(sendCommitQueueGithubStatus(j.env, pr, message.GithubStateFailure, "can't update commit queue item", patchDoc.Id.Hex()))
```

Comment:
I think I was thinking for all the failure modes it would be confusing for the user to get a link to the patch. In other words, if we're going to dequeue the patch the patch link probably isn't relevant for the user.

Reply:
I don't really think the patch link is relevant for any of these 🤷

Reply:
I thought it's good for the success case, though.